### PR TITLE
Release v0.6.0

### PR DESCRIPTION
**Pull request by ConnorStoneAstro**
Making PR now so we can track merge compatability

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+See the [Scientific Python Developer Guide][spc-dev-intro] for a detailed
+description of best practices for developing scientific packages.
+
+[spc-dev-intro]: https://scientific-python-cookie.readthedocs.io/guide/intro
```
Comment (ConnorStoneAstro) [inline .github/CONTRIBUTING.md:4]: This link appears to be broken @lsetiawan

Comment (lsetiawan) [reply .github/CONTRIBUTING.md:4]: @uwcdc could you see what's going on here? Thanks!

Comment (uwcdc) [reply .github/CONTRIBUTING.md:4]: @ConnorStoneAstro the Scientific Python link changed with the development version and needed to be updated in the CONTRIBUTING.md file to the following stable version:

https://scientific-python-cookie.readthedocs.io/en/stable/

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          # Install tar gz and check import
+          python -m pip uninstall --yes caustics
+          python -m pip install --extra-index-url https://test.pypi.org/simple --upgrade --pre --no-binary=:all: caustics
+          python -c "import caustics; print(caustics.__version__)"
+          echo "=== Done testing source tar file ==="
+
+  publish:
+    needs: [dist, test-built-dist]
+    name: Publish to PyPI
+    environment: pypi
```
Comment (ConnorStoneAstro) [inline .github/workflows/cd.yml:86]: Do we need to make a `pypi` environment for caustics? I think this will fail the workflow if it doesn't exist

Comment (lsetiawan) [reply .github/workflows/cd.yml:86]: No this will be created by Github. 😄

Comment (lsetiawan) [reply .github/workflows/cd.yml:86]: Though, it is a good catch as on the pypi end, we have to specify the correct environment.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+]
+
+[tool.hatch.metadata.hooks.requirements_txt]
+files = ["requirements.txt"]
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.hooks.vcs]
+version-file = "src/caustics/_version.py"
```
Comment (ConnorStoneAstro) [inline pyproject.toml:56]: I can't find the `src/caustics/_version.py` file, is this created dynamically?

Comment (lsetiawan) [reply pyproject.toml:56]: Yes it is! And this is in the `.gitignore` since this changes automatically by the auto versioning

**Review by AlexandreAdam — APPROVED** [review]

**Review by lsetiawan — APPROVED** [review]
LGTM

**Pull request merged**